daemon/api_download.go

```diff
@@ -68,38 +108,154 @@ func postSnapDownload(c *Command, r *http.Request, user *auth.UserState) Respons
 }
 
 func streamOneSnap(c *Command, action snapDownloadAction, user *auth.UserState) Response {
-	actions := []*store.SnapAction{{
-		Action:       "download",
-		InstanceName: action.SnapName,
-		Revision:     action.Revision,
-		CohortKey:    action.CohortKey,
-		Channel:      action.Channel,
-	}}
-	sars, err := getStore(c).SnapAction(context.TODO(), nil, actions, user, nil)
+	secret, err := downloadTokensSecret(c)
 	if err != nil {
-		return errToResponse(err, []string{action.SnapName}, InternalError, "cannot download snap: %v")
+		return InternalError(err.Error())
+	}
+	theStore := getStore(c)
+
+	var ss *snapStream
+	if action.ResumeToken == "" {
+		var info *snap.Info
+		actions := []*store.SnapAction{{
+			Action:       "download",
+			InstanceName: action.SnapName,
+			Revision:     action.Revision,
+			CohortKey:    action.CohortKey,
+			Channel:      action.Channel,
+		}}
+		sars, err := theStore.SnapAction(context.TODO(), nil, actions, user, nil)
+		if err != nil {
+			return errToResponse(err, []string{action.SnapName}, InternalError, "cannot download snap: %v")
+		}
+		if len(sars) != 1 {
+			return InternalError("internal error: unexpected number %v of results for a single download", len(sars))
+		}
+		info = sars[0].Info
+
+		ss, err = newSnapStream(action.SnapName, info, secret)
+		if err != nil {
+			return InternalError(err.Error())
+		}
+	} else {
+		var err error
```

Comment:
Is this variable declaration important? Why wouldn't we use `:=` on the next line? If it *is* important, can you please add a comment explaining it.

Reply:
we are trying to assign to ss from outside, if we use := we get a local version of it. It's not a nice expect of go this, but that's how it is